tests/handlers/tests.py

```diff
@@ -33,12 +33,16 @@ def test_middleware_initialized(self):
         self.assertIsNotNone(handler._request_middleware)
 
     def test_bad_path_info(self):
-        """Tests for bug #15672 ('request' referenced before assignment)"""
+        """
+        Non-UTF-8 encoded path will populate PATH_INFO with an URL-encoded path
```

Comment:
Looks good. You could use this style for the docstring: A non-UTF-8 path populates ... and produces...